deleter/ec2.go

```diff
@@ -1154,12 +1168,106 @@ func (rd *EC2NatGatewayDeleter) DeleteResources(cfg *DeleteConfig) error {
 			return err
 		}
 
+	}
+
+	if cfg.DryRun {
+		return nil
+	}
+
+	// If we don't wait until nat gateways are deleted, EIP disassociation/release
+	// and customer gateway disassociation/deletion will fail
+	fmt.Println("Waiting for EC2 NAT Gateways to delete...")
+	deletedNGWs, aliveNGWs, err := rd.waitUntilNatGatewaysDeleted(ngws)
```

Comment:
I would agree with you, but I've run into several issues deleting resources (EIP's and IGW's) relating to NAT gateways not being fully deleted. Instances are another example of a resources that must be deleted before proceeding, and which have a `wait` function.